### PR TITLE
Dl.daily comment fix up

### LINKED ISSUE #54: Report/Comment Modal Implementation
**Description: We would like to expand on [#Issue 6](https://github.com/Code-4-Community/breaktime-frontend/issues/6#issue-1574692595) and make slight design tweaks & add some pop up modals to allow supervisors to make edits & comment on employee timesheets.**

Figma Designs: 

Image

Click through this prototype for more details: 
https://www.figma.com/proto/Yl4q9gMJndfkAVaTxVtWce/Breaktime-Timesheet?type=design&node-id=320-8026&scaling=min-zoom&page-id=0%3A1&starting-point-node-id=320%3A8026&mode=design

Main Components: 

- Implement a way to flag incorrect entries reported by an employee when hovered upon (maybe even have a popover that tells employees to refer to the feedback section in the employee view)
- Have an "add report" and "add comment button" when there are no comments or reports initially
- When the "add report" button is clicked, a modal should include a dropdown of 5 reasons why the report was created: "late arrival", "early arrival", "early departure", "late departure", or "no show". If an option besides no show is selected, supervisors should be prompted to enter a start and end time for that particular shift
- When the "add comment" button is selected, a modal should populate with a text body to type a comment and submit. 
- When a comment or report is added have a way to view them and add consecutive comments (refer to figma design)

### PR DESCRIPTION
**Pull request by circlegov**
ℹ️ Issue
Closes #54 

📝 Description
Briefly list the changes made to the code:

Added comment and report modals to match figma

✔️ Testing
What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves.

image
image
image
image
image
image

Provide screenshots of any new components, styling changes, or pages.

🏕️ (Optional) Future Work / Notes
Did you notice anything ugly during the course of this ticket? Any bugs, design challenges, or unexpected behavior? Write it down so we can clean it up in a future ticket!

### CODE REVIEW

File: src/components/TimeCardPage/CellTypes/CommentModals/ShowCommentModal.tsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      const user = useContext(UserContext);
+  
+      const handleRemarkChange = (e) => {
+        setRemark(e.target.value);
+      };
+  
+      const handleSubmit = () => {
+        // TODO: reuse comment validation
+        setComments([...comments, createNewComment(user, CommentType.Comment, remark)]);
+        alert(`Your ${CommentType.Comment} has been submitted!`);
```
Comment (aditya161216) [inline src/components/TimeCardPage/CellTypes/CommentModals/ShowCommentModal.tsx:172]: I think you can remove this alert, might look better without it

File: src/components/TimeCardPage/CellTypes/CommentModals/ShowReportModal.tsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+              (report) => (
+                <HStack>
+                  {/* TODO: add UserDisplay card once pr merged in*/}
+                  <Editable
+                    isDisabled={!isEditable}
+                    defaultValue={report.Content}
+                    onSubmit={(value) => saveEditedComment(setReports, reports, CommentType.Report, report, createNewComment(user, CommentType.Report, value))}
+                  >
+                    <EditablePreview />
+                    {` ${moment.unix(report.Timestamp).format("h:mm")}`}
```
Comment (aditya161216) [inline src/components/TimeCardPage/CellTypes/CommentModals/ShowReportModal.tsx:121]: Can you include AM or PM as well

File: src/components/TimeCardPage/CellTypes/CommentModals/ShowCommentModal.tsx (hunk 1 above)
Comment (aditya161216) [inline src/components/TimeCardPage/CellTypes/CommentModals/ShowCommentModal.tsx:172]: I think you can remove this alert, might look better without it

Comment (circlegov) [reply src/components/TimeCardPage/CellTypes/CommentModals/ShowCommentModal.tsx:172]: just there as a placeholder, should be replaced by a toast later on. I'll add a comment to have a toast there instead

**Review by kayleewuhu — APPROVED** [review]
LGTM

**Pull request merged**